dataherald/sql_database/models/types.py

```diff
@@ -93,16 +108,25 @@ def validate_uri(cls, input_string):
         match = re.match(pattern, input_string)
         if not match:
             raise InvalidURIFormatError(f"Invalid URI format: {input_string}")
+        return match.group(1)
+
+    @classmethod
+    def set_dialect(cls, input_string):
+        for dialect in SupportedDialects:
+            if dialect.value in input_string:
+                return dialect.value
+        return None
 
     @validator("connection_uri", pre=True, always=True)
-    def connection_uri_format(cls, value: str):
+    def connection_uri_format(cls, value: str, values):
         fernet_encrypt = FernetEncrypt()
         try:
-            fernet_encrypt.decrypt(value)
-            return value
+            decrypted_value = fernet_encrypt.decrypt(value)
+            dialect_prefix = cls.validate_uri(decrypted_value)
```

Comment:
so the `validate_uri` returns a dialect? Seems like it does 2 things and it might be confusing. In general the validators should return a boolean flag indicating that the validation rule passed or not